internal/pkg/cli/app_logs.go

```diff
@@ -0,0 +1,384 @@
+	}
+
+	if o.envName == "" {
+		envs, err = o.storeSvc.ListEnvironments(o.ProjectName())
+		if err != nil {
+			return fmt.Errorf("list environments: %w", err)
+		}
+		if len(envs) == 0 {
+			return fmt.Errorf("no environments found in project %s", color.HighlightUserInput(o.ProjectName()))
+		}
+	} else {
+		env, err := o.storeSvc.GetEnvironment(o.ProjectName(), o.envName)
+		if err != nil {
+			return fmt.Errorf("get environment: %w", err)
+		}
+		envs = []*archer.Environment{env}
+	}
+
+	appEnvs := make(map[string]appEnv)
+	var appEnvNames []string
+	for _, appName := range appNames {
+		for _, env := range envs {
+			if err := o.initCwLogsSvc(o, env); err != nil {
+				return err
+			}
+			deployed, err := o.cwlogsSvc[env.Name].LogGroupExists(fmt.Sprintf(logGroupNamePattern, o.ProjectName(), env.Name, appName))
+			if err != nil {
+				return err
+			}
+			if !deployed {
+				continue
+			}
+			appEnv := appEnv{
+				appName: appName,
+				envName: env.Name,
+			}
+			appEnvName := appEnv.String()
+			appEnvs[appEnvName] = appEnv
+			appEnvNames = append(appEnvNames, appEnvName)
+		}
+	}
+	if len(appEnvNames) == 0 {
+		return fmt.Errorf("no deployed applications found in project %s", color.HighlightUserInput(o.ProjectName()))
+	}
+
+	// return if only one deployed app found
+	if len(appEnvNames) == 1 {
+		log.Infof("Only found one deployed app, defaulting to: %s\n", color.HighlightUserInput(appEnvNames[0]))
+		o.appName = appEnvs[appEnvNames[0]].appName
+		o.envName = appEnvs[appEnvNames[0]].envName
+
+		return nil
+	}
+
+	appEnvName, err := o.prompt.SelectOne(
+		fmt.Sprintf(applicationLogAppNamePrompt),
+		applicationLogAppNameHelpPrompt,
+		appEnvNames,
+	)
+	if err != nil {
+		return fmt.Errorf("select deployed applications for project %s: %w", o.ProjectName(), err)
+	}
+	o.appName = appEnvs[appEnvName].appName
+	o.envName = appEnvs[appEnvName].envName
+
+	return nil
+}
+
+func (o *appLogsOpts) retrieveProjectNames() ([]string, error) {
+	projs, err := o.storeSvc.ListProjects()
+	if err != nil {
+		return nil, fmt.Errorf("list projects: %w", err)
+	}
+	projNames := make([]string, len(projs))
+	for ind, proj := range projs {
+		projNames[ind] = proj.Name
+	}
+	return projNames, nil
+}
+
+func (o *appLogsOpts) retrieveAllAppNames() ([]string, error) {
+	apps, err := o.storeSvc.ListApplications(o.ProjectName())
+	if err != nil {
+		return nil, fmt.Errorf("list applications for project %s: %w", o.ProjectName(), err)
+	}
+	appNames := make([]string, len(apps))
+	for ind, app := range apps {
+		appNames[ind] = app.Name
+	}
+
+	return appNames, nil
+}
+
+func (o *appLogsOpts) outputLogs(logs []*cloudwatchlogs.Event) error {
+	if o.shouldOutputJSON {
+		for _, log := range logs {
+			data, err := log.JSONString()
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(o.w, data)
+		}
+	} else {
+		for _, log := range logs {
+			fmt.Fprintf(o.w, log.HumanString())
+		}
+	}
+	return nil
+}
+
+func (o *appLogsOpts) parseSince(since int64) *int64 {
```

Comment:
this function is a bit odd because it takes an `int64` and returns a pointer to also an `int64` :P Instead:
```go
func (o *appLogsOpts) parseSince() int64 {
  sec := int64(o.since.Round(time.Second).Seconds())
  timestamp := time.Now().Add(time.Duration(-sec) * time.Second)
  return timestamp.Unix() * 1000
}
```